sys-auth/elogind/files/elogind-255.5-musl.patch

```diff
@@ -0,0 +1,196 @@
+diff --git a/src/basic/cgroup-util.c b/src/basic/cgroup-util.c
+index 90cfeb5..b88ac7c 100644
+--- a/src/basic/cgroup-util.c
++++ b/src/basic/cgroup-util.c
+@@ -1535,7 +1535,11 @@ int cg_path_get_session(const char *path, char **ret_session) {
+         if (e == n)
+                 return -ENXIO;
+ 
++#if defined(__GLIBC__)
+         start = strndupa(e, n - e);
++#else
++        start = strndup(e, n - e);
++#endif
+         start = cg_unescape(start);
+ 
+         if (!start[0])
+diff --git a/src/basic/errno-util.h b/src/basic/errno-util.h
+index 27804e6..63a83ed 100644
+--- a/src/basic/errno-util.h
++++ b/src/basic/errno-util.h
+@@ -10,12 +10,20 @@
+ /* strerror(3) says that glibc uses a maximum length of 1024 bytes. */
+ #define ERRNO_BUF_LEN 1024
+ 
++// https://github.com/chriskohlhoff/asio/blob/efdc25ab99786101351a5afb39f01dfaf0781401/asio/include/asio/impl/error_code.ipp#L188
++// Helper function to adapt the result from glibc's variant of strerror_r.
++#if defined(__GLIBC__)
++static const char* strerror_result(const char* s, const char*) { return s; }
++#else
++static const char* strerror_result(int, const char* s) { return s; }
++#endif
++
+ /* Note: the lifetime of the compound literal is the immediately surrounding block,
+  * see C11 §6.5.2.5, and
+  * https://stackoverflow.com/questions/34880638/compound-literal-lifetime-and-if-blocks
+  *
+  * Note that we use the GNU variant of strerror_r() here. */
+-#define STRERROR(errnum) strerror_r(abs(errnum), (char[ERRNO_BUF_LEN]){}, ERRNO_BUF_LEN)
++#define STRERROR(errnum) strerror_result(strerror_r(abs(errnum), (char[ERRNO_BUF_LEN]){}, ERRNO_BUF_LEN), (char[ERRNO_BUF_LEN]){})
+ 
+ /* A helper to print an error message or message for functions that return 0 on EOF.
+  * Note that we can't use ({ … }) to define a temporary variable, so errnum is
+diff --git a/src/basic/missing_prctl.h b/src/basic/missing_prctl.h
+index 7d9e395..5320169 100644
+--- a/src/basic/missing_prctl.h
++++ b/src/basic/missing_prctl.h
+@@ -1,7 +1,9 @@
+ /* SPDX-License-Identifier: LGPL-2.1-or-later */
+ #pragma once
+ 
++#if defined(__GLIBC__)
+ #include <linux/prctl.h>
++#endif
+ 
+ /* 58319057b7847667f0c9585b9de0e8932b0fdb08 (4.3) */
+ #ifndef PR_CAP_AMBIENT
+diff --git a/src/basic/rlimit-util.c b/src/basic/rlimit-util.c
+index 091c111..59bdc35 100644
+--- a/src/basic/rlimit-util.c
++++ b/src/basic/rlimit-util.c
+@@ -428,11 +428,7 @@ int rlimit_nofile_safe(void) {
+         rl.rlim_max = MIN(rl.rlim_max, (rlim_t) read_nr_open());
+         rl.rlim_cur = MIN((rlim_t) FD_SETSIZE, rl.rlim_max);
+         if (setrlimit(RLIMIT_NOFILE, &rl) < 0)
+-#ifdef __GLIBC__ /// To be compatible with musl-libc, elogind uses an (uintmax_t) cast.
```

Comment:
Why is this being removed?

Reply:
```
../elogind-255.5/src/basic/rlimit-util.c:434:47: error: format '%llu' expects argument of type 'long long unsigned int', but argument 7 has type 'long unsigned
nt' [-Werror=format=]
  434 |                 return log_debug_errno(errno, "Failed to lower RLIMIT_NOFILE's soft limit to " RLIM_FMT ": %m", (uintmax_t)rl.rlim_cur);
      |                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                  ~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                                                 |
      |                                                                                                                 long unsigned int
../elogind-255.5/src/basic/log.h:232:86: note: in definition of macro 'log_full_errno_zerook'
  232 |                         ? log_internal(_level, _e, PROJECT_FILE, __LINE__, __func__, __VA_ARGS__) \
      |                                                                                      ^~~~~~~~~~~
../elogind-255.5/src/basic/log.h:272:41: note: in expansion of macro 'log_full_errno'
  272 | #define log_debug_errno(error, ...)     log_full_errno(LOG_DEBUG,   error, __VA_ARGS__)
      |                                         ^~~~~~~~~~~~~~
../elogind-255.5/src/basic/rlimit-util.c:434:24: note: in expansion of macro 'log_debug_errno'
  434 |                 return log_debug_errno(errno, "Failed to lower RLIMIT_NOFILE's soft limit to " RLIM_FMT ": %m", (uintmax_t)rl.rlim_cur);
      |                        ^~~~~~~~~~~~~~~
```

Reply:
Please also see:
https://github.com/elogind/elogind/issues/269 and
https://github.com/elogind/elogind/commit/53e84063a1016b129544886867552b61e92ad228

Reply:
i.e. we can pull in that patch instead.

Reply:
@hhoffstaette @thesamesam That commit seems to present in 255.5

https://github.com/elogind/elogind/commit/7c4eb382991575235aa0b1dc91cea4211287a43d

Reply:
https://github.com/elogind/elogind/pull/288